### PR TITLE
LESQ-560: Specialist Lesson Overview Page

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: {owa_music_year}

- Adds SpecialistLessonOverview query, page, schema and test file
- Adds method to alter keys from snake_case to camelCase and test file

## Issue(s)

Fixes #LESQ-560

## How to test

1. Go to https://deploy-preview-2310--oak-web-application.netlify.thenational.academy
2. Click on \_\_\_\_\_\_
3. You should see \_\_\_\_\_\_

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/node-lib/curriculum-api-2023/queries/specialistLessonOverview/specialistLessonOverview.schema.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  threadSlug: z.string(),
+  threadTitle: z.string(),
+});
+
+type Threads = z.infer<typeof threads>;
+
+const combined_programme_fields = z.object({
+  subject: z.string(),
+  subject_slug: z.string(),
+  developmentstage: z.string(),
```
Comment (njblore) [inline src/node-lib/curriculum-api-2023/queries/specialistLessonOverview/specialistLessonOverview.schema.ts:59]: developmentstage and phase need to be nullable, see `specialistLessonDownloadRawSchema`

File: src/node-lib/curriculum-api-2023/queries/specialistLessonOverview/specialistLessonOverview.schema.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    misconceptions_and_common_mistakes: z
+      .array(misconceptions_and_commom_mistakes_schema)
+      .nullable(),
+    equipment_and_resources: z
+      .array(lesson_equipment_and_resources_schema)
+      .nullable()
+      .optional(),
+    teacher_tips: z.array(teacher_tips_schema).nullable(),
+    content_guidance: z.array(content_guidance_schema).nullable(),
+    supervision_level: z.string(),
```
Comment (njblore) [inline src/node-lib/curriculum-api-2023/queries/specialistLessonOverview/specialistLessonOverview.schema.ts:90]: this is nullable also

**Review by njblore — CHANGES_REQUESTED** [review]
I see what you mean about the snake case types! 😵‍💫 just some issues with the schema I can see, I'd test it out with different types of lesson to make sure there are no other issues as well

**Review by jack-skerrett-oak — APPROVED** [review]
Lesson overview pages still seem to be overviewing

File: src/node-lib/curriculum-api-2023/queries/specialistLessonOverview/specialistLessonOverview.schema.ts (hunk 1 above)
Comment (k-huggs) [reply src/node-lib/curriculum-api-2023/queries/specialistLessonOverview/specialistLessonOverview.schema.ts]: Fixed ✅

**Review by njblore — APPROVED** [review]

**Pull request merged**